### DISCUSSION TITLE
Added object not visible until selected

### DISCUSSION THREAD

**Discussion by fpolli** [fabricjs/fabric.js#7990] [Q&A]
I have a confounding issue where I have a Canvas object, I load an Image, then I create a Rect, set some options to it, add it, and it does not show. If I then click on its area, it will come visible.

I have searched and found several references to a similar problem on SO and Google groups, and have tried calling renderAll explicitly, tried calling it with a setTimeout pause of 1000, tried using object.set to set dirty and call renderAll. I have tried using requestRenderAll instead of renderAll, and nothing works. It just will not appear until clicked.

**Comment by fpolli** [comment]
If you can't beat them, select them. canvas.setActiveObject(myrect) after the add did the trick

**Comment by ShaMan123** [comment]
Without a reproduction there is no way to understand/help/investigate

**Reply by ShaMan123** [reply]
Try `canvas.skipOffscreen=false`
